assume/common/scenario_loader.py

```diff
@@ -542,19 +552,37 @@ def run_learning(world: World, inputs_path: str, scenario: str, study_case: str)
         world.learning_role.buffer = buffer
         world.learning_role.episodes_done = episode
 
+
         if episode + 1 >= world.learning_role.learning_starts:
             world.learning_role.turn_off_initial_exploration()
 
         world.run()
         actors_and_critics = world.learning_role.extract_actors_and_critics()
-        avg_reward = world.output_role.get_episode_sum_reward()
-        path = world.learning_config["load_learned_path"]
-        if best_reward < avg_reward:
-            world.learning_role.save_params(directory=path)
-
-        if episode == world.learning_role.training_episodes - 1:
-            world.learning_role.save_params(directory=path)
-
+        
+        if episode % world.learning_config.get("validation_episodes_interval", 5) == 0:
```

Comment:
This should be episode+1, otherwise it will run a validation at the very start as well

Reply:
addressed